contrib/nixos/default.nix

```diff
@@ -3,13 +3,18 @@
 with lib;
 
 let
+  
   gosh = pkgs.buildGoModule {
     name = "gosh";
 
-    src = lib.cleanSource ./.;
-
+    src = pkgs.fetchFromGitHub {
+      owner = "oxzi";
+      repo = "gosh";
+      rev = "89b614613232e1fa89db5112c3a72aaf7c160916";
+      sha256 = "sha256-Krd3YP9YFZ0f6PsM4xH2IN9RiZo9oxVKRUmp8ITpsl4=";
+    };
     # TODO: One has to configure this one.
-    vendorSha256 = "0000000000000000000000000000000000000000000000000000";
+    vendorHash = "sha256-TcrUG8k/SxDqXgnzYqS1hWNAnAHswbWuBoQ+nSYdBBk=";
```

Comment:
I explicitly have not defined a revision as this will lead to breakages and, thus, the end user has to explicitly define which revision should be used. Otherwise, this have to be bumped for every other commit.

Reply:
At least this change should be done before NixOS 24.05, because:

> trace: warning: `vendorSha256` is deprecated. Use `vendorHash` instead